### PR TITLE
Add I18N Support

### PR DESCRIPTION
**Pull request by suzuki**
## About

- Add I18N support
    - `en` and `ja`

### Not include:

- Admin UI
- React components

### User Setting 

- Users can setting own language

### CODE REVIEW

**Comment by sotarok** [comment]
Thank you for the great PR, I'll check it soon.

File: lib/crowi/express-init.js [hunk 1]
```diff
@@ -12,10 +12,27 @@ module.exports = function(crowi, app) {
...
+    , i18next        = require('i18next')
+    , i18nFsBackend  = require('i18next-node-fs-backend')
+    , i18nSprintf    = require('i18next-sprintf-postprocessor')
+    , i18nMiddleware = require('i18next-express-middleware')
     , env            = crowi.node_env
     , middleware     = require('../util/middlewares')
     ;
 
+  i18next
+    .use(i18nMiddleware.LanguageDetector)
```
Comment (sotarok) [inline lib/crowi/express-init.js]: Are you sure how `i18nMiddleware.LanguageDetector` detect user language?
I'm using Mac and browser with English mode and found that the UI was still Japanese though (on [review app](https://crowi-dev-pr-143.herokuapp.com/)).

**Comment by suzuki** [comment]
Added some features:

- User's language setting
    - Add to `language` field to User model
- Translate login (sign in) view

**Comment by suzuki** [comment]
Fixed conflict by merge `master`

File: locales/en-US/translation.json [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  "page_register": {
+    "notice": {
+      "restricted": "New registration to this wiki is restricted.",
+      "restricted_defail": "In order to start using it, approval by the administrator is necessary after new registration.",
+      "google_account_continue": "Please decide user ID, name and password and continue registration."
+    },
+    "form_help": {
+      "email": "In this wiki, only the following e-mail address can be registered.",
+      "password": "Please enter at least 6 letters, numbers, alphanumeric characters, symbols etc for password.",
+      "user_id": "The user ID is used for the URL of the user page etc. Only half-width alphanumeric characters and some symbols can be used."
```
Comment (Hidsm) [inline locales/en-US/translation.json]: The URL of pages you create will contain your User ID. Your User ID can consist of letters, numbers, and some symbols.

File: locales/en-US/translation.json [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    }
+  },
+
+  "page_me": {
+    "form_help": {
+      "google_connect1": "With Google Connect, you can sign in with your Google Account.",
+      "google_connect2": "Only Google Apps accounts with the following email addresses are connectable Google accounts:",
```
Comment (Hidsm) [inline locales/en-US/translation.json]: You must have a Google Apps account with an @mercari.com email address to connect to this wiki

File: locales/en-US/translation.json [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+      "google_disconnect1": "If disconnet Google, you can not login by Google Authentication.",
```
Comment (Hidsm) [inline locales/en-US/translation.json]: If you disconnect your Google account, you will be unable to sign in using Google Authentication

File: locales/en-US/translation.json [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+      "google_disconnect2": "After disconnect, you can login using email and password."
```
Comment (Hidsm) [inline locales/en-US/translation.json]: After disconnecting your Google account, you can sign in normally using your email and password

File: locales/en-US/translation.json [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+    }
+  },
+  "page_me_apitoken": {
+    "notice": {
+      "apitoken_issued": "API Token is not issued.",
+      "update_token1": "With updating, the new API Token will be generated.",
```
Comment (Hidsm) [inline locales/en-US/translation.json]: You can update to generate a new API Token.

File: locales/en-US/translation.json [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+      "update_token2": "You have to update the processes using the current API Token."
```
Comment (Hidsm) [inline locales/en-US/translation.json]: You will need to update the API Token in any existing processes.

File: locales/en-US/translation.json [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+    },
+    "form_help": {
+    }
+  },
+
+  "Password": "Password",
+  "Password Setting": "Password Setting",
```
Comment (Hidsm) [inline locales/en-US/translation.json]: Password Settings

File: locales/en-US/translation.json [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+  "Set new Password": "Set new Password",
+  "Update Password": "Update Password",
+  "Current password": "Current password",
+  "New password": "New password",
+  "for check": "for check",
```
Comment (Hidsm) [inline locales/en-US/translation.json]: Re-enter new password

File: locales/en-US/translation.json [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+  "Please set a password": "Please set a password",
+  "You can login email and password": "You can login <code>%s</code> and password",
```
Comment (Hidsm) [inline locales/en-US/translation.json]: You can sign in with <code>%s</code> and password

File: locales/en-US/translation.json [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+
+  "API Setting": "API Setting",
```
Comment (Hidsm) [inline locales/en-US/translation.json]: API Settings

File: locales/en-US/translation.json [hunk 12]
```diff
@@ -0,0 +1,164 @@
...
+  "API Token Setting": "API Token Setting",
```
Comment (Hidsm) [inline locales/en-US/translation.json]: API Token Settings

File: locales/en-US/translation.json [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
+      }
+  },
+
+  "Rename page": "Rename page",
+  "New page name": "New page name",
+  "Current page name": "Current page name",
+  "Redirect": "Redirect",
+  "modal_rename": {
+    "help": {
+      "redirect": "Redirect to the new page if someone access <code>%s</code>"
```
Comment (Hidsm) [inline locales/en-US/translation.json]: Redirect to new page if someone accesses <code>%s</code>

File: locales/en-US/translation.json [hunk 14]
```diff
@@ -0,0 +1,164 @@
...
+    }
+  },
+
+  "modal_help": {
+      "basic": {
+          "title": "基本的な機能",
```
Comment (Hidsm) [inline locales/en-US/translation.json]: Basics

File: locales/en-US/translation.json [hunk 15]
```diff
@@ -0,0 +1,164 @@
...
+          "body1": "表示される画面には、「一覧ページ」と「ページ」の2種類があります",
```
Comment (Hidsm) [inline locales/en-US/translation.json]: There are 2 types of pages: List pages (showing lists of links to other pages) and normal pages.

File: locales/en-US/translation.json [hunk 16]
```diff
@@ -0,0 +1,164 @@
...
+          "body2": "スラッシュ <code>/</code> で終わるページは、その階層の一覧ページとなります。",
```
Comment (Hidsm) [inline locales/en-US/translation.json]: Pages that end with a slash <code>/</code> are List pages for anything following the slash.

File: locales/en-US/translation.json [hunk 17]
```diff
@@ -0,0 +1,164 @@
...
+          "body3": "ページでの変更はすべて記録されています。History からそのページの過去の状態を見ることができます。"
```
Comment (Hidsm) [inline locales/en-US/translation.json]: You can view older versions of a page from the History tab. Any changes made will be stored here.

File: locales/en-US/translation.json [hunk 18]
```diff
@@ -0,0 +1,164 @@
...
+      },
+      "tips": {
+          "title": "編集のコツ",
```
Comment (Hidsm) [inline locales/en-US/translation.json]: Quick Tips on Editing

File: locales/en-US/translation.json [hunk 19]
```diff
@@ -0,0 +1,164 @@
...
+          "body1": "文章の <strong>構造</strong> を意識しましょう。本を書くように、内容と文脈を整理してセクション・サブセクション...と構造的に書くと、わかりやすく他人に伝わりやすいページがになります。"
```
Comment (Hidsm) [inline locales/en-US/translation.json]: Use sections and subsections to make it easier for your friends to read your pages.

File: locales/en-US/translation.json [hunk 20]
```diff
@@ -0,0 +1,164 @@
...
+      },
+      "markdown": {
+          "title": "記法"
```
Comment (Hidsm) [inline locales/en-US/translation.json]: Markdown Rules

File: locales/en-US/translation.json [hunk 21]
```diff
@@ -0,0 +1,164 @@
...
+
+  "Sign in": "Sign in",
+  "Sign up is here": "Sign up",
+  "Sign in is here": "Sign in",
+  "Registration": "Registration",
+  "Register by Google Account": "Register by Google Account",
+  "Sign in by Google Account": "Sign in by Google Account",
+  "Register with this Google Account": "Register with this Google Account",
+  "Example": "Example",
+  "Taro Yamada": "Taro Yamada",
```
Comment (sotarok) [inline locales/en-US/translation.json]: 👍

**Pull request merged**